lib/graphql/schema/build_from_definition.rb

```diff
@@ -69,6 +68,14 @@ def build(document, default_resolve: DefaultResolve, using: {}, interpreter: tru
             end
           end
 
+          types = GraphQL::Schema::BUILT_IN_TYPES.merge(types) do |key, built_in_type, existing_type|
```

Comment:
@rmosolgo had to add some merging logic in here to deal with late bound types. Otherwise when `orphan_types` was called a default scalar (which is late bound) would never be resolved. Not sure if you can think of a better solution.

Reply:
I didn't understand at first ... but I pulled the branch and gave it a try without the code. Now I see what you mean: if it's only _referenced_ in the given document, but not defined, then we can't ever resolve the late-bound type. I added an exploratory comment and rejiggered the code a bit so that it made more sense to simpleminded maintainer of this library 😰 .